lib/chat/getConversations.js

```diff
@@ -28,7 +28,13 @@ exports.func = (args) => {
     }
   }).then((res) => {
     if (res.statusCode !== 200) {
-      throw new Error('You are not logged in')
+      const body = JSON.parse(res.body) || {}
+      if (body.errors && body.errors.length > 0) {
+        const errors = body.errors.map((e) => {
```

Comment:
I've noticed this pattern in a few places in the library. Is a body.errors array a standard thing across the Roblox API? Could this be moved to a utility function?

Reply:
Also I couldn't agree more with your rationale for the change: Would it be worth defining a noblox-specific error class (i.e. one that extends Error) that could contain extra info like status code, the roblox error text and an optional description for extra context that could be set by the library?
This is again a design decision so open to discussion. Either way I think this represents a semver major change so would need to form part of v5.

Reply:
> I've noticed this pattern in a few places in the library. Is a body.errors array a standard thing across the Roblox API? Could this be moved to a utility function?

It generally is, yes. Maybe in another PR

Reply:
> Also I couldn't agree more with your rationale for the change: Would it be worth defining a noblox-specific error class (i.e. one that extends Error) that could contain extra info like status code, the roblox error text and an optional description for extra context that could be set by the library?
> This is again a design decision so open to discussion. Either way I think this represents a semver major change so would need to form part of v5.

Yes, definitely.